### PR TITLE
add explaination on model.eval()

### PR DESCRIPTION
**Pull request by zhiqiu**
As the title, add explaination on `model.eval()` in dygraph mode, since some users may forget to call `model.eval()` when they are actually doing `evaluation`.

Related issue in Paddle repo, #[21685](https://github.com/PaddlePaddle/Paddle/issues/21685).

### CODE REVIEW

**Comment by zhiqiu** [comment]
image

**Review by sneaxiy — APPROVED** [review]
LGTM.

File: doc/fluid/user_guides/howto/dygraph/DyGraph.md [hunk 1]
```diff
@@ -688,7 +688,9 @@ with fluid.dygraph.guard():
 
 ## 模型评估
 
-当我们需要在DyGraph模式下利用搭建的模型进行预测任务，可以使用`YourModel.eval()`接口，在之前的手写数字识别模型中我们使用`mnist.eval()`来启动预测模式（我们默认在`fluid.dygraph.guard()`上下文中是训练模式），在预测的模式下，DyGraph将只会执行前向的预测网络，而不会进行自动求导并执行反向网络：
+当我们需要在DyGraph模式下利用搭建的模型进行预测任务，请在`fluid.dygraph.guard()`上下文中调用一次`YourModel.eval()`接口来启动预测模式。例如，在之前的手写数字识别模型中我们可以使用`mnist.eval()`来启动预测模式。需要显示地调用`YourModel.eval()`启动预测模式的原因是，我们默认在`fluid.dygraph.guard()`上下文中是训练模式，训练模式下DyGraph在运行前向网络的时候会自动求导，添加反向网络；而在预测时，DyGraph将只需要执行前向的预测网络，不需要进行自动求导并执行反向网络。
```
Comment (phlrain) [inline doc/fluid/user_guides/howto/dygraph/DyGraph.md]: 这个启动预测模式，改成 切换到预测模式 ，，是不是好一点，因为如果要训练的话，还要切换回来

Comment (zhiqiu) [reply doc/fluid/user_guides/howto/dygraph/DyGraph.md]: 好的

**Comment by zhiqiu** [comment]
image

**Review by saxon-zh — APPROVED** [review]
LGTM

**Pull request merged**